python-package/lightgbm/dask.py

```diff
@@ -306,9 +306,15 @@ def _train(
         'voting_parallel'
     }
     if params["tree_learner"] not in allowed_tree_learners:
-        _log_warning('Parameter tree_learner set to %s, which is not allowed. Using "data" as default' % params['tree_learner'])
+        _log_warning(f'Parameter tree_learner set to {params["tree_learner"]}, which is not allowed. Using "data" as default')
         params['tree_learner'] = 'data'
 
+    if params['tree_learner'] not in {'data', 'data_parallel'}:
+        _log_warning(
+            f'Support for tree_learner {params["tree_learner'"} in lightgbm.dask is experimental and may break in a future release. \n'
+            'Use "data" for a stable, well-tested interface.'
+        )
+
```

Comment:
This warning was removed from `lightgbm` in #4088. Did you mean to add it back in here, or is this just the result of 
 solving a merge conflict?

Please remove it from this pull request and update this branch to the latest `master`.